src/petals/server/throughput.py

```diff
@@ -104,21 +103,43 @@ def measure_network_rps(config: BloomConfig) -> float:
     return network_rps
 
 
-def measure_device_rps(device: str, config: BloomConfig, layer_index: int = 0, n_steps: int = 500) -> float:
+def measure_compute_rps(
+    config: BloomConfig,
+    device: torch.device,
+    dtype: torch.dtype,
+    *,
+    load_in_8bit: bool,
+    layer_index: int = 0,
+    n_steps: int = 500,
+) -> float:
     with torch.inference_mode():
-        block = BloomBlock(config, layer_index).to(device)
+        block = BloomBlock(config, layer_index).to(dtype)
+        if load_in_8bit:
+            block = replace_8bit_linear(block)
+        block = block.to(device)
+
         cache = None
         elapsed = 0
-        for i in range(n_steps):
-            dummy_input = torch.randn(1, 1, config.hidden_size, device=device)
-            alibi = build_alibi_tensor(i + 1, config.num_attention_heads, dtype=torch.float32, device=device)
+        for step in range(n_steps + 1):
+            dummy_input = torch.randn(1, 1, config.hidden_size, device=device, dtype=dtype)
+            alibi = build_alibi_tensor(step + 1, config.num_attention_heads, device=device, dtype=dtype)
 
             start_time = time.perf_counter()
             _, cache = block.forward(dummy_input, alibi=alibi, use_cache=True, layer_past=cache)
-            elapsed += time.perf_counter() - start_time
+            if step >= 1:  # Skip the 1st step to exclude the initialization time
+                elapsed += time.perf_counter() - start_time
         device_rps = n_steps / elapsed
 
-    device_name = f"{torch.cuda.get_device_name(0)} GPU" if device == "cuda" else "CPU"
-    logger.info(f"Compute throughput ({device_name}): {device_rps:.2f} RPS")
-
+    logger.info(
+        f"Compute throughput ({_get_device_name(device)}, {_get_dtype_name(dtype, load_in_8bit)}): "
+        f"{device_rps:.2f} RPS"
+    )
     return device_rps
+
+
+def _get_device_name(device: torch.device) -> str:
+    return f"{torch.cuda.get_device_name(0)} GPU" if device == "cuda" else "CPU"
```

Comment:
```suggestion
    device = torch.device(device)  # compatibility with string device names
    return f"{torch.cuda.get_device_name(device.index)} GPU" if device.type == "cuda" else "CPU"
```

Nit
![image](https://user-images.githubusercontent.com/3491902/205278456-9aaf2573-6862-4bf1-9e1e-68c933a07b89.png)
![image](https://user-images.githubusercontent.com/3491902/205278694-d670b409-35d3-4982-9219-2765f84ada74.png)